src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -231,13 +267,33 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
   result->peer_identity =
       static_cast<char*>(gpr_zalloc(service_account.size + 1));
   memcpy(result->peer_identity, service_account.data, service_account.size);
-  upb::Arena arena;
+  upb::Arena rpc_protocol_arena;
   bool serialized = grpc_gcp_rpc_protocol_versions_encode(
-      peer_rpc_version, arena.ptr(), &result->rpc_versions);
+      peer_rpc_version, rpc_protocol_arena.ptr(), &result->rpc_versions);
   if (!serialized) {
     gpr_log(GPR_ERROR, "Failed to serialize peer's RPC protocol versions.");
     return TSI_FAILED_PRECONDITION;
   }
+  upb::Arena context_arena;
+  grpc_gcp_AltsContext* context = grpc_gcp_AltsContext_new(context_arena.ptr());
+  grpc_gcp_AltsContext_set_application_protocol(context, application_protocol);
+  grpc_gcp_AltsContext_set_record_protocol(context, record_protocol);
+  grpc_gcp_AltsContext_set_security_level(context, 2);
+  grpc_gcp_AltsContext_set_peer_service_account(context, service_account);
```

Comment:
Could you please rename it to `peer_service_account` to make it distinguishable from `local_service account`?